src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -116,9 +116,9 @@ bool shared_node_optimization(const shared_ptr<Model>& model) {
             const auto& target_inputs = output.get_target_inputs();
             if (target_inputs.size() <= 1)
                 continue;  // nothing to optimize
-            unordered_map<Node::type_info_t, vector<Node*>> type_to_node;
+            unordered_map<Node::type_info_t, vector<std::shared_ptr<Node>>> type_to_node;
```

Comment:
We have some concerns about this fix.
this will be investigated separately so as not to block this PR